src/transformStateWithClones.js

```diff
@@ -1,13 +1,39 @@
 'use strict';
 
+/* Fix */
 /**
  * @param {Object} state
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const results = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        stateCopy = { ...stateCopy, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        stateCopy = { ...stateCopy };
+
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+        break;
+
+      case 'clear':
+        stateCopy = {};
+        break;
+    }
+
+    results.push({ ...stateCopy }); // Важливо: пушимо копію!
```

Comment:
Please remove the comments from the code, especially if it is in Ukrainian. These types of comments are usually left by AI.